api/minekube/gate/v1/gate_service.proto

```diff
@@ -191,4 +228,281 @@ message Player {
+  PROXY_MODE_LITE = 2;
+}
+
+// GetStatusResponse contains proxy runtime metadata.
+message GetStatusResponse {
+  string version = 1;
+  ProxyMode mode = 2;
+  int32 players = 3;
+  int32 servers = 4;
+}
+
+// ConfigFormat specifies the serialization format for config payloads.
+enum ConfigFormat {
+  CONFIG_FORMAT_UNSPECIFIED = 0;
+  CONFIG_FORMAT_JSON = 1;
+  CONFIG_FORMAT_YAML = 2;
+}
+
+// GateConfig represents the root configuration structure
+message GateConfig {
+  APIConfig api = 1;
+  HealthServiceConfig health_service = 2;
+  ConnectConfig connect = 3;
+  JavaConfig config = 4;
+}
+
+// APIConfig represents the Gate API configuration
+message APIConfig {
+  bool enabled = 1;
+  string bind = 2;
+}
+
+// HealthServiceConfig represents the health service configuration
+message HealthServiceConfig {
+  bool enabled = 1;
+  string bind = 2;
+}
+
+// ConnectConfig represents the Connect network configuration
+message ConnectConfig {
+  bool enabled = 1;
+  string name = 2;
+  bool allow_offline_mode_players = 3;
+}
+
+// JavaConfig represents the main Java edition configuration
+message JavaConfig {
+  string bind = 1;
+  bool online_mode = 2;
+  ForwardingConfig forwarding = 3;
+  StatusConfig status = 4;
+  map<string, string> servers = 5;
+  repeated string try = 6;
+  map<string, StringList> forced_hosts = 7;
+  bool accept_transfers = 8;
+  bool bungee_plugin_channel_enabled = 9;
+  LiteConfig lite = 10;
+  // Add other commonly used fields as needed
+}
+
+// ForwardingConfig represents player info forwarding settings
+message ForwardingConfig {
+  string mode = 1;
+  string velocity_secret = 2;
+  string bungee_guard_secret = 3;
+}
+
+// StatusConfig represents status response settings
+message StatusConfig {
+  string motd = 1;
+  int32 show_max_players = 2;
+  string favicon = 3;
+  bool log_ping_requests = 4;
+  bool announce_forge = 5;
+}
+
+// LiteConfig represents Gate Lite mode configuration
+message LiteConfig {
+  bool enabled = 1;
+  repeated LiteRoute routes = 2;
+}
+
+// StringList helper for repeated string values in maps
+message StringList {
+  repeated string values = 1;
+}
+
+// GetConfigRequest is the request for GetConfig method.
+// If format is unspecified JSON will be returned.
+message GetConfigRequest {
+  ConfigFormat format = 1;
+}
+
+// GetConfigResponse contains the serialized config payload.
+message GetConfigResponse {
+  ConfigFormat format = 1;
+  string payload = 2;
+}
+
+// ValidateConfigRequest is the request for ValidateConfig method.
+// Use either yaml_config for YAML configurations or json_config for structured JSON configurations.
+message ValidateConfigRequest {
+  oneof config {
+    string yaml_config = 1;       // YAML configuration as string
+    GateConfig json_config = 2;   // JSON configuration as structured protobuf
+  }
+}
+
+// ValidateConfigResponse contains validation warnings when the config is valid.
+message ValidateConfigResponse {
+  repeated string warnings = 1;
```

Comment:
We also have an array of errors right? Or does the rpc fail then. How would we report multiple errors